### PR TITLE
BigQuery keyfile_dict profile mapping use env vars for sensitive fields

### PR DESCRIPTION
**Pull request by jbandoro**
## Description
I noticed that if `GoogleCloudServiceAccountDictProfileMapping` is used for profile mapping, the full keyfile json is logged in the UI. See example below:
image

This PR hides the sensitive private key fields (`private_key` and `private_key_id`) as env vars so that they will not show up in the logs, see below:
image

Would appreciate feedback since unlike other profiles, the `secret_fields` here are nested within the `keyfile_json` field so I had to override `BaseProfileMapping.env_vars` here to make it work.
## Related Issue(s)

None

## Breaking Change?

None

## Checklist

- [x] I have made corresponding changes to the documentation (if required)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: cosmos/profiles/bigquery/service_account_keyfile_dict.py [hunk 1]
```diff
@@ -48,12 +53,27 @@ def profile(self) -> dict[str, Any | None]:
...
-            return keyfile_json
-        keyfile_json = json.loads(keyfile_json)
-        if isinstance(keyfile_json, dict):
-            return keyfile_json
+            keyfile_json_dict = keyfile_json
         else:
-            raise ValueError("keyfile_json cannot be loaded as a dict.")
+            keyfile_json_dict = json.loads(keyfile_json)
+            if not isinstance(keyfile_json_dict, dict):
+                raise ValueError("keyfile_json cannot be loaded as a dict.")
```
Comment (tatiana) [inline cosmos/profiles/bigquery/service_account_keyfile_dict.py:64]: Could we add a test case covering this scenario?

File: cosmos/profiles/bigquery/service_account_keyfile_dict.py [hunk 2]
```diff
@@ -48,12 +53,27 @@ def profile(self) -> dict[str, Any | None]:
...
+
+        for field in self.secret_fields:
+            value = keyfile_json_dict.get(field)
+            if value is None:
+                raise CosmosValueError(f"Could not find a value in service account json field: {field}.")
```
Comment (tatiana) [inline cosmos/profiles/bigquery/service_account_keyfile_dict.py:69]: Could we add a test case covering this scenario?

**Review by tatiana — COMMENTED** [review]
This looks great, @jbandoro ! Thanks a lot for your contribution.

Sorry for the misleading coverage comments (PR #768 should fix this).

Can we add tests covering the exceptions being raised? I'm happy to approve/merge the PR once this is done.

File: cosmos/profiles/bigquery/service_account_keyfile_dict.py (hunk 2 above)
Comment (jbandoro) [reply cosmos/profiles/bigquery/service_account_keyfile_dict.py:69]: Added a test to check that the error is raised as expected.

File: cosmos/profiles/bigquery/service_account_keyfile_dict.py (hunk 1 above)
Comment (jbandoro) [reply cosmos/profiles/bigquery/service_account_keyfile_dict.py]: Added a test covering this, and updated it raise a `CosmosValueError` to be consistent.

File: cosmos/profiles/bigquery/service_account_keyfile_dict.py (hunk 2 above)
Comment (tatiana) [reply cosmos/profiles/bigquery/service_account_keyfile_dict.py:69]: Thanks for addressing this so quickly, @jbandoro !

**Review by tatiana — APPROVED** [review]
Thank you very much for addressing the feedback, @jbandoro , and fixing this issue.

**Pull request merged**